src/components/filter/Filter.tsx

```diff
@@ -1,130 +1,211 @@
-    }
-    if (tagsSet.has('pipelines') === true) {
-      tagsSet.clear();
-      tagsSet.add('pipelines');
-    }
-    tagArray = Array.from(tagsSet);
-    for (let i = 0; i < tagArray.length; i++) {
-      if (i === 0 && tagArray[i] !== 'task') {
-        str += '?tags=';
+  // / dummy function for refreshing component
+  const ddff=() =>{
+    setX((x) => x+1);
+  };
+  //  formation of filter url for fetching task and pipelines
+  const filterurl=(e:any) =>{
+    let typeurl='all';
+    let verifiedurl='all';
+    let tagsurl = '';
+
+    status.forEach((it:any) => {
+      if (it.id === e.target.id) {
+        it.isChecked=e.target.checked;
       }
-      str = `${str + tagArray[i]}|`;
+    },
+    );
+    setStatus(status);
+    ddff();
+    const temptype = status.slice(0, 2);
+    const tempverified=status.slice(2, 3);
+    const temptags=status.slice(3);
+    if (tempverified[0].isChecked === true) {
+      verifiedurl='true';
     }
-    fetch(`${API_URL}/tasks${str}`)
-        .then((res) => res.json())
-        .then((data) => {
-          store.dispatch({type: 'FETCH_TASK_SUCCESS', payload: data});
-        });
-  };
-  const addTag = (e: any) => {
-    if (tagsSet.has(e.target.value) === false) {
-      tagsSet.add(e.target.value);
-    } else {
-      tagsSet.delete(e.target.value);
+    if (temptype[0].isChecked === true) {
+      typeurl='task';
     }
-    displaytask();
-  };
-  const newtags =tags.slice(0, i);
-  // / jsx element for show tags
-  const showTags:any=
-  newtags.map((it: any) => (
-    <div key = {it} style={{marginBottom: '0.5em'}}>
-      <Checkbox
-        style={{width: '1.2em', height: '1.2em'}}
-        label={it.name[0].toUpperCase()+it.name.slice(1)}
-        value={it.name}
-        id={it.id}
-        onClick={addTag}
-        aria-label="uncontrolled checkbox example"
-      />
-    </div>
-  ));
-  const tagSize:number = tags.length;
-  const [lessTags, setLessTags] = React.useState(tagSize);
-  // /  for display more tags
-  const moreTags=(e:any) =>{
-    if (i >= newtags.length ) {
-      setI(tagSize);
-      // temp = tagSize;
-      setLessTags(tagSize);
-      setToggle('seeLess');
-    } else {
-      setI(newtags.length+7);
+    if (temptype[1].isChecked === true) {
+      typeurl ='pipeline';
     }
-    if (e.target.text.match('seeLess')) {
-      setI(lessTags-(13));
-      setToggle('seeMore');
+    if ((temptype[0].isChecked === true) &&
+       (temptype[1].isChecked === true)) {
+      typeurl = 'all';
     }
+    temptags.forEach((it:any) =>{
+      if (it.isChecked === true) {
+        tagsurl+=it.value+'|';
+      }
+      if (it.isChecked === true) {
+        setClear('ClearAll');
+      }
+    });
+
+    fetch(`${API_URL}/resources/${typeurl}/${verifiedurl}?tags=${tagsurl} `)
+        .then((resp) => resp.json())
+        .then((data) =>{
+          store.dispatch(
+              {
+                type: FETCH_TASK_SUCCESS,
+                payload: data,
+              },
```

Comment:
Can we break this component into smaller components and as you are using dispatch can have  some connected components as needed